beacon/beacon.go

```diff
@@ -243,8 +243,13 @@ func (pd *ProtocolDriver) Start(ctx context.Context) {
 }
 
 func (pd *ProtocolDriver) UpdateBeacon(epoch types.EpochID, beacon types.Beacon) {
-	// TODO: implement
-	pd.logger.With().Info("received beacon update", epoch, beacon)
+	pd.mu.Lock()
+	defer pd.mu.Unlock()
+	if err := beacons.AddOverwrite(pd.cdb, epoch, beacon); err != nil {
+		pd.logger.With().Error("failed to persist fallback beacon", epoch, beacon, log.Err(err))
+	}
```

Comment:
this error handling is not sound. i suggest to return an error and fail the node if it is not possible to persist something (or retry, until it is possible to persist it)

it is definitely not an option just to continue

Reply:
changed to fail the node.